pkg/agent/route/route_windows.go

```diff
@@ -129,10 +135,135 @@ func (c *Client) listRoutes() (map[string]*netroute.Route, error) {
 		if rt.LinkIndex != c.nodeConfig.GatewayConfig.LinkIndex {
 			continue
 		}
-		if rt.DestinationSubnet.IP.IsLoopback() {
+		// Only process IPv4 route entries in the loop.
+		if rt.DestinationSubnet.IP.To4() == nil {
 			continue
 		}
-		rtMap[rt.DestinationSubnet.String()] = &rt
+		// Retrieve the route entries with destination using global unicast only. This is because the function
+		// "GetNetRoutesAll" also returns the entries of loopback, broadcast, and multicast, which are
+		// added by the system when adding a new IP on the interface. Since removing those route entries might
+		// introduce the host networking issues, ignore them from the list.
+		if rt.DestinationSubnet.IP.IsGlobalUnicast() {
+			// Windows adds an active route for the local broadcast address automatically when a new IP address
+			// is configured on the interface. This route entry should be ignored in the result.
+			if rt.DestinationSubnet.IP.Equal(util.GetLocalBroadcastIP(rt.DestinationSubnet)) {
+				continue
+			}
+			rtMap[rt.DestinationSubnet.String()] = &rt
+		}
 	}
 	return rtMap, nil
 }
+
+const (
+	inboundFirewallRuleName  = "Antrea: accept packets from local pods"
+	outboundFirewallRuleName = "Antrea: accept packets to local pods"
+)
+
+type fwRuleAction string
```

Comment:
How about moving the generic code to pkg/agent/util/winfirewall like what we did for iptables? @wenyingd @jianjuns

Reply:
Sure, I will update.